amplify/function/BusinessLogic/AssignUsersToTeams/handler.ts

```diff
@@ -0,0 +1,139 @@
+import { Amplify } from "aws-amplify";
+import { generateClient } from "aws-amplify/data";
+import type { AppSyncResolverHandler } from "aws-lambda";
+
+import type { Schema } from "../../../data/resource";
+import type { ModelUserConnection } from "./graphql/API";
+import { updateUser } from "./graphql/mutations";
+import { getTeam, getUser } from "./graphql/queries";
+
+const MAX_TEAM_MEMBERS = 6;
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: process.env.AMPLIFY_DATA_GRAPHQL_ENDPOINT as string,
+        region: process.env.AWS_REGION,
+        defaultAuthMode: "iam",
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: process.env.AWS_ACCESS_KEY_ID as string,
+            secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY as string,
+            sessionToken: process.env.AWS_SESSION_TOKEN as string,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  },
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+type ResolverArgs = { userId: string; teamId: string };
+
+type ResolverResult = {
+  body: { value: string };
+  statusCode: number;
+  headers: { "Content-Type": string };
+};
+
+export const handler: AppSyncResolverHandler<
+  ResolverArgs,
+  ResolverResult
+> = async (event, _) => {
+  try {
+    const user = (
+      await client.graphql({
+        query: getUser,
+        variables: {
+          id: event.arguments.userId,
+        },
+      })
+    ).data.getUser;
+
+    const team = (
+      await client.graphql({
+        query: getTeam,
+        variables: {
+          id: event.arguments.teamId,
+        },
+      })
+    ).data.getTeam;
+
+    if (team == null) {
+      return {
+        body: { value: "Error: Team does not exist" },
+        statusCode: 404,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+
+    if (user == null) {
+      return {
+        body: { value: "Error: User does not exist" },
+        statusCode: 404,
+        headers: { "Content-Type": "application/json" },
+      };
+    }
+
+    if (await user.team) {
```

Comment:
Because of the way we made the graphQL query, these all return as part of the main await so we no longer need to await any of these. We can remove this await